19.03.5.sh

```diff
@@ -511,6 +515,39 @@ do_install() {
 			)
 			exit 0
 			;;
+		sles)
+			(
+			set -x
+			$sh_c 'zypper refresh'
+			docker_pkg=$(zypper se -n -s docker | grep -E "\sdocker\s.*${docker_version//\./\\.}.*$architecture" | sort --sort=v -k3 | tail -1 | cut -d '|' -f4 | cut -d ' ' -f2)
+			if [ -z "$docker_pkg" ]; then
+				semverParse $docker_version
+				trunc_docker_version=$major.$minor
+				available_docker_pkgs=$(zypper se -n -s docker | grep -E "\sdocker\s.*${trunc_docker_version//\./\\.}.*$architecture" | sort --sort=v -k3 | cut -d '|' -f4 | cut -d ' ' -f2 | cut -d '_' -f1)
+				if [ -z "$available_docker_pkgs" ]; then
+					cat >&2 <<-EOF
+					SLES $dist_version does not support any patches for docker $trunc_docker_version for $architecture.
```

Comment:
Not sure why I missed this before, this implies the OS does not support it but this is not necessarily true as the repo can be missing to provide the docker package. We should just tell the user there was no package found for Docker, and to check enabled repositories.

Reply:
Updated in the latest commit!

Reply:
Updated in the latest commit!